src/coordsim/traffic_predictor/traffic_predictor.py

```diff
@@ -34,28 +35,14 @@ def predict_traffic(self):
             sfc_ids = [*sfc_ids]  # New unpacking trick in python 3.5+
             sfc = sfc_ids[0]
             ingress_sf = self.params.sfc_list[sfc][0]
-
-            inter_arr_mean = None
-            if self.params.deterministic_arrival:
-                inter_arr_mean = self.params.predicted_inter_arr_mean[node_id]
-            else:
-                inter_arr_times = []
-                # Poisson arrival -> exponential distributed inter-arrival time
-                for _ in range(self.params.run_duration):
-                    inter_arr_times.append(random.expovariate(lambd=1.0/self.params.predicted_inter_arr_mean[node_id]))
-                    # If the duration of the inter_arr_times reaches the run duration, assume enough flows generated
-                    # for a run
-                    # This is to simulate waiting before generating another flow
-                    if sum(inter_arr_times) >= self.params.run_duration:
-                        break
-                inter_arr_mean = np.mean(inter_arr_times)
-            # Calculate flow count for each ingress node
-            flow_dr = 0.0
-            number_of_flows = self.params.run_duration / inter_arr_mean
-
-            # Predict data rate for expected number of flows
-            for _ in range(ceil(number_of_flows)):
-                flow_dr += np.random.normal(self.params.flow_dr_mean, self.params.flow_dr_stdev)
+            flow_dr = 0
+            # DDPG Weirdly calls `init()` twice which causes the second (more important) init call to
```

Comment:
We've noticed this multiple times and we should debug it at some point. I don't get, why we do 2x init. Could you open an issue (backlog or todo)?

Reply:
How do you currently handle this case? You just predict 0?

Reply:
```python
if self.last_dr_idx[node_id] == len(self.params.flow_dr_list[node_id]):
    self.last_dr_idx[node_id] = 0`
```

I check here for when we are at the end of the list and reset the index to 0. Since we are only at the actual end of the list only when this two calls happen, I simply reset the index to 0.

Reply:
Ah because, you already iterate to the end of the list in the first init call? Then you reset it for the 2nd init call. And in all future calls, you won't iterate until the very end?